pennylane_lightning/src/tests/Test_KernelMap.cpp

```diff
@@ -137,3 +137,67 @@ TEST_CASE("Test KernelMap functionalities", "[KernelMap]") {
             Util::LightningException, "does not exist");
     }
 }
+
+TEST_CASE("Test KernelMap is consistent in extreme usecase", "[KernelMap]") {
+    using Gates::GateOperation;
+    using Gates::KernelType;
+    using EnumKernelMap =
+        OperationKernelMap<Gates::GateOperation>::EnumKernelMap;
+    auto &instance = OperationKernelMap<Gates::GateOperation>::getInstance();
+
+    const auto num_qubits = std::vector<size_t>{4, 6, 8, 10, 12, 14, 16};
+    const auto threadings =
+        std::vector<Threading>{Threading::SingleThread, Threading::MultiThread};
+    const auto memory_models = std::vector<CPUMemoryModel>{
+        CPUMemoryModel::Unaligned, CPUMemoryModel::Aligned256,
+        CPUMemoryModel::Aligned512};
+
+    std::random_device rd;
+
+    std::vector<EnumKernelMap> records;
+
+    records.push_back(instance.getKernelMap(12, Threading::SingleThread,
+                                            CPUMemoryModel::Aligned256));
+
+#ifdef _OPENMP
+#pragma omp parallel default(none)                                             \
+    shared(instance, records, rd, num_qubits, threadings, memory_models)
+#endif
+    {
+        std::mt19937 re;
+
+#ifdef _OPENMP
+#pragma omp critical
+#endif
+        { re.seed(rd()); }
+
+        std::uniform_int_distribution<size_t> num_qubit_dist(
+            0, num_qubits.size() - 1);
+        std::uniform_int_distribution<size_t> threading_dist(
+            0, threadings.size() - 1);
+        std::uniform_int_distribution<size_t> memory_model_dist(
+            0, memory_models.size() - 1);
+
+        std::vector<EnumKernelMap> res;
+
+#ifdef _OPENMP
+#pragma omp for
+#endif
+        for (size_t i = 0; i < 1024 * 8; i++) {
```

Comment:
All good with this number?

Reply:
I just made this as a `constexpr` variable instead of hard coding here. Runtime-wise, it is fine (less than 0.1 sec).

Reply:
Sounds good -- thanks!